src/linear_algebra/orientation.lean

```diff
@@ -402,6 +433,49 @@ begin
     simp [h₁, h₂]
 end
 
+/-- If the index type has cardinality equal to the finite dimension, an orientation equals the
+negation of another orientation if and only if they are not equal. -/
+lemma ne_iff_eq_neg (x₁ x₂ : orientation R M ι) (h : fintype.card ι = finrank R M) :
```

Comment:
Do you think it's worth having copies of this lemma for `[strong_ring_condition R]` where `h` can be filled with some variant of `dim_eq_card_basis`?

Reply:
I'm not clear what you are suggesting here. There are versions both with an explicit basis involved in the lemma (`orientation_ne_iff_eq_neg`) and with that `h` instead of specifying a choice of basis; the latter relies on a field for the existence of a basis (but I expect could be generalized to `module.free` if `linear_algebra.finite_dimensional` gets so generalized in future).

Reply:
I think I'm getting confused here, ignore me.